app/routes.py

```diff
@@ -1,2 +1,325 @@
+def update_video_by_id(video_id):
+    video = Video.query.get(video_id)
+    request_data = request.get_json()
+    
+    if video is None:
+        return make_response({"details": f"Video id number: {video_id} not found"}, 404)
+    
+    if not request_data.get("title") or type(request_data.get("title")) != str:
+        return make_response({"details": "Please enter a valid value for title field."}, 400)
+    elif not request_data.get("release_date") or type(request_data.get("release_date")) != str:
+        return make_response({"details": "Please enter a valid value for release_date field."}, 400)
+    elif not request_data.get("total_inventory")  or type(request_data.get("total_inventory")) != int:
+        return make_response({"details": "Please enter a valid value for total_inventory field."}, 400)
+    
+    request_data = request.get_json()
+
+    video.title = request_data["title"]
+    video.release_date = request_data["release_date"]
+    video.total_inventory = request_data["total_inventory"]
+
+    db.session.commit()
+
+    currently_rented_out = len(video.rentals)
+    total = video.total_inventory
+    available = total - currently_rented_out
+
+    response_body = {
+            "id": video.id,
+            "title": video.title,
+            "release_date": video.release_date,
+            "total_inventory": video.total_inventory,
+            "available_inventory": available
+    }
+
+    return make_response(response_body, 200)
+
+@video_bp.route("/<video_id>", methods=["DELETE"], strict_slashes=False)
+def delete_video(video_id):
+    video = Video.query.get(video_id)
+
+    if video is None:
+        return make_response({"details": f"Video id number: {video_id} not found"}, 404)
+
+    db.session.delete(video)
+    db.session.commit()
+
+    return make_response({"id": f"{video.id}"}, 200)
+
+
+#Rental routes and many to many routes
+
+@rental_bp.route("/check-out", methods=["POST"],strict_slashes=False)
+def check_out_video():
+
+    request_body = request.get_json()
+
+    customer_id = request_body["customer_id"]
+    video_id = request_body["video_id"]
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+
+    if not request_body:
+        return make_response("", 404)
+
+    if video is None:
+        return make_response({"details": f"Video id number: {video_id} not found"}, 404)
+    
+    if customer is None:
+        return make_response({"details": f"Video id number: {customer_id} not found"}, 404)
+
+    now = datetime.now()
+    due = timedelta(days=+7)
+    due_date = now + due   
+
+    new_rental = Rental(
+        customer_id = request_body["customer_id"],
+        video_id = request_body["video_id"], 
+        due_date = due_date
+
+    )
+
+    video.total_inventory = video.total_inventory - 1
+
+    db.session.add(new_rental)
+    db.session.commit()
+
+    checked_out_count = len(customer.rentals)
```

Comment:
I suggest moving this whole business logic of renting a video into a model function.  This helps dry up your routes and creates separation between the handling requests with the business function of checking out a video to a customer.